### PR TITLE
logger: invoke logger_compatible_adjust before run_tikv (#11651)

### LINKED ISSUE #11789: tikv have not logs saved in k8s
### What version of TiKV are you using?
/ # ./tikv-server -V
TiKV
Release Version: 5.4.0-alpha
Edition: Community
Git Commit Hash: 99b3436
Git Commit Branch: heads/refs/tags/v5.4.0-nightly
UTC Build Time: 2022-01-04 01:15:55
Rust Version: rustc 1.56.0-nightly (2faabf579 2021-07-27)
Enable Features: jemalloc mem-profiling portable sse test-engines-rocksdb cloud-aws cloud-gcp cloud-azure
Profile: dist_release

### What operating system and CPU are you using?
8core 16G

### Steps to reproduce
no matter

### What did you expect?
tikv logs can be saved 

### What did happened?
tikv have not logs saved in k8s 
image

### PR DESCRIPTION
**Pull request by ethercflow**
close #11789 

Signed-off-by: Wenbo Zhang <ethercflow@gmail.com>

### What is changed and how it works?

Close #11789  

What's Changed:

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- PR to update `pingcap/tidb-ansible`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)

Side effects

- None

### Release note 

```release-note
None
```

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -2787,6 +2787,59 @@ impl TiKvConfig {
         Ok(())
     }
 
+    // As the init of `logger` is very early, this adjust needs to be separated and called
```
Comment (5kbpers) [inline src/config.rs:2790]: Can we add a test for this case?

**Review by 5kbpers — APPROVED** [review]

**Comment by ethercflow** [comment]
This PR is to fix a bug: when `log-file` is not set in the command line, `log-file` is set in the configuration file, but `log.file.filename` is not configured, and the log information is printed to the terminal.

The reason is that the log module uses log parameters for initialization earlier than the compatibility adjustment of the parameters.

**Review by gengliqi — APPROVED** [review]

File: src/config.rs [hunk 2]
```diff
@@ -2780,6 +2780,59 @@ impl TiKvConfig {
...
+    // immediately after parsing the command line.
+    pub fn logger_compatible_adjust(&mut self) {
+        let default_tikv_cfg = TiKvConfig::default();
+        let default_log_cfg = LogConfig::default();
+        if self.log_level != default_tikv_cfg.log_level {
+            println!("deprecated configuration, log-level has been moved to log.level");
```
Comment (BusyJay) [inline src/config.rs:2789]: You should not use `println!`.

If the logger is initialized, then you should use `info!` instead, if it's not, `eprintln!` should be used instead.

Comment (ethercflow) [reply src/config.rs:2789]: Sorry, I missed email notification. I'll submit another PR to fix this. Thanks!

**Pull request merged**